apps/users/src/app/app.controller.ts

```diff
@@ -1381,4 +1381,26 @@ export class UserController {
 
     return '';
   }
+  @CastcleBasicAuth()
+  @Get('me/advertise')
+  async listAds(
+    @Auth() { account }: Authorizer,
+    @Body() paginationQuery: PaginationQuery
+  ) {
+    let response = [];
+    const adsCampaigns = await this.adsService.listAds(
+      account,
+      paginationQuery
+    );
+    if (!adsCampaigns) return { payload: null };
+    for (const adsCampaign of adsCampaigns) {
+      const prepareAds =
+        await this.adsService.transformAdsCampaignToAdsResponse(adsCampaign);
+      response = [...response, prepareAds];
+    }
```

Comment:
ใช้ `Array.map` แทนครับ

```suggestion
    const ads = await Promise.all(
        adsCampaigns.map(adsCampaign=> this.adsService.transformAdsCampaignToAdsResponse(
            adsCampaign
        )))
```